### PR TITLE
removed compare macros from asr builder

### PR DESCRIPTION
**Pull request by parth121101**
fixes: #3644

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I'll review this in a while

File: src/libasr/asr_builder.h [hunk 1]
```diff
@@ -537,46 +537,62 @@ class ASRBuilder {
...
+        return EXPR(ASR::make_IntegerCompare_t(al, loc, x, ASR::cmpopType::GtE, y, logical, nullptr));
+    }
+    inline ASR::expr_t* iGt(ASR::expr_t* x, ASR::expr_t* y) {
+        return EXPR(ASR::make_IntegerCompare_t(al, loc, x, ASR::cmpopType::Gt, y, logical, nullptr));
+    }
 
     #define ArraySize_1(x, dim) EXPR(make_ArraySize_t_util(al, loc, x, dim,     \
         int32, nullptr))
     #define ArraySize_2(x, dim, t) EXPR(make_ArraySize_t_util(al, loc, x, dim,  \
         t, nullptr))
```
Comment (Pranavchiku) [inline src/libasr/asr_builder.h:562]: Please remove these two as well.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
A small change requested, once done this will be good to merge.

File: src/libasr/asr_builder.h (hunk 1 above)
Comment (parth121101) [reply src/libasr/asr_builder.h]: Done

**Review by Pranavchiku — APPROVED** [review]
Looks good to me, thanks @parth121101 !

**Comment by Pranavchiku** [comment]
When you rebase there should be no commit like https://github.com/lfortran/lfortran/pull/3657/commits/0ecec603717856e0a79b7031ec32447f95ac2792

**Pull request merged**